program/src/lib.rs

```diff
@@ -37,3 +37,51 @@ pub fn get_violation_report_address(
         &id(),
     )
 }
+
+struct ViolationReportAddress<'a> {
```

Comment:
not sure if me trying to reuse some code made this better or worse, happy to delete this if you think worse.

Reply:
Moved it to a separate file here f9ab5127270cddaea677bc00a4b5685f882db49d
After removing the system program from CloseViolationReport, we actually don't need it there, so there's only one consumer. I've kept it here in case we need it in the future.